src/clients/kvno/t_kvno.py

```diff
@@ -0,0 +1,76 @@
+from k5test import *
+
+realm = K5Realm()
+
+def check_cache(ccache, expected_services):
+    # Fetch the klist output and skip past the header.
+    lines = realm.run([klist, '-c', ccache]).splitlines()
+    lines = lines[4:]
+
+    # For each line not beginning with an indent, match against the
+    # expected service principals.
+    svcs = {x: True for x in expected_services}
+    for l in lines:
+        if not l.startswith('\t'):
+            svcprinc = l.split()[4]
+            if svcprinc in svcs:
+                del svcs[svcprinc]
+            else:
+                fail('unexpected service princ ' + svcprinc)
+
+    if svcs:
+        fail('services not found in klist output: ' + ' '.join(keys((svcs))))
+
+
+mark('no options')
+realm.run([kvno, realm.user_princ], expected_msg='user@KRBTEST.COM: kvno = 1')
+check_cache(realm.ccache, [realm.krbtgt_princ, realm.user_princ])
+
+mark('-e')
+msgs = ('etypes requested in TGS request: des3-cbc-sha1',
+        '/KDC has no support for encryption type')
+realm.run([kvno, '-e', 'des3-cbc-sha1', realm.host_princ],
```

Comment:
How would you feel about using a more futureproof enctype here?

Reply:
I'll switch to a Camellia enctype; we're less likely to remove those in the forseeable future than des3 (although I have considered it; I don't think there's much use).

(aes-sha2 isn't future-proof for a different reason: we might add it to the default supported_enctypes, causing kvno to succeed where we currently expect it to fail.)